enable/savage/svg/pathdata.py

```diff
@@ -40,7 +42,13 @@ class CaselessPreservingLiteral(CaselessLiteral):
 
     def __init__(self, matchString):
         super().__init__(matchString.upper())
-        self.name = "'%s'" % matchString
+
+        pyparsing_version = parse(pyparsing_version_str)
+        if pyparsing_version.major == 2:
+            self.name = "'%s'" % matchString
+        elif pyparsing_version.major == 3:
+            self.set_name("'%s'" % matchString)
+
         self.errmsg = "Expected " + self.name
```

Comment:
Checking the library version every time seems a bit too much (and could be brittle). I think the better solution here is to use `setName` in the old version and `set_name` in the new version. Something like:
```python
quoted_name = f"'{matchString}'"
if hasattr(self, "set_name"):
    # New version 3+ API
    self.set_name(quoted_name)
else:
    # Old API
    self.setName(quoted_name)
```
Note that pyparsing sets `self.errmsg` in both versions, so this makes things slightly simpler here.

[Old version](https://github.com/pyparsing/pyparsing/blob/pyparsing_2.4.7/pyparsing.py#L1452-L1465)
[New version](https://github.com/pyparsing/pyparsing/blob/0352555d968f9952800f0728383de8e1f9526e1e/pyparsing/core.py#L1789-L1800)

Reply:
I wasn't sure about using the old `setName` API specifically because it was doing more than just updating the `name` attribute so I wasn't sure if it was fit for purpose.

Let's go with your solution if it is in fact fit for purpose.

Reply:
Other than the
```python
if __diag__.enable_debug_on_named_expressions:
    self.setDebug()
return self
```
part, it's exactly the same as the code in our `__init__` method.